node/dataSources/catalog.ts

```diff
@@ -136,21 +164,66 @@ export class CatalogDataSource extends IODataSource {
     return this.http.getRaw<T>(`/proxy/catalog${url}`, config)
   }
 
+  private collectionsUrl = (query: string) =>
```

Comment:
(⛏  I would rather inline this function)

Reply:
It really depends on the prettier, but I would rather remove this functions to a utils or just outside the class, since you don't need `this` anywhere